### PR TITLE
Chatbot updates

### PR DESCRIPTION
**Pull request by MattBroach**
Upgrades to channels 2.0, moves main event loop to asyncio.  Could use more tests, but there are some internal structure upgrades that I'm going to be making in the nearish future, so I'll likely expand the test coverage with that.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,33 +11,53 @@ run `python setup.py install` to install the library and set up the command line
...
+The server requires that the `server`, `nickname`, and `application` properties be set. The `application` should be an import string pointing to the location of your app's ASGI application. Hence, if your app was named `myapp`, contained an ASGI filed called `asgi.py`, and your ASGI application is named `my_application`, you could start the server by running:
 
 ```
-channels-chatbot -s 'irc.freenode.net' -n 'my_irc_nickname' -c 'myapp.asgi:my_channel_layer'
+channels-irc -s 'irc.freenode.net' -n 'my_irc_nickname' -a 'myapp.asgi:my_application'
 ```
 
-You can also set these values using the env varialbes `CHANNELS_CHATBOT_SERVER`, `CHANNELS_CHATBOT_NICKNAME`, and `CHANNELS_CHATBOT_LAYER`.
+You can also set these values using the env varialbes `CHANNELS_IRC_SERVER`, `CHANNELS_IRC_NICKNAME`, and `CHANNELS_IRC_LAYER`.
```
Comment (bd) [inline README.md]: typo: varialbes

File: channels_irc/cli.py [hunk 2]
```diff
@@ -101,19 +101,21 @@ def run(self, args):
             format="%(asctime)-15s %(levelname)-8s %(message)s",
         )
 
-        if not any([args.channel_layer, args.server, args.nickname]):
+        if not any([args.application, args.server, args.nickname]):
```
Comment (bd) [inline channels_irc/cli.py]: I could be mistaken, but if as the error message says, `--application`, `--server`, and `--nickname` are required, then `any()` should be `all()`. `not any()` is only true if none of the arguments are `True`. So it seems like either the exception message or the function call is incorrect

```
>>> any([True, True, False])
True
>>> any([False, False, False])
False
>>> all([True, True, False])
False
```

File: channels_irc/cli.py [hunk 3]
```diff
@@ -123,11 +125,28 @@ def run(self, args):
             ircname=args.realname
         )
 
+        logger.error("THIS IS AN ERROR LOG")
```
Comment (bd) [inline channels_irc/cli.py]: Looks like debugging remainders?

File: channels_irc/cli.py (hunk 2 above)
Comment (MattBroach) [reply channels_irc/cli.py]: 100% correct.  Good catch.

**Pull request merged**